github/Workflow.py

```diff
@@ -162,10 +170,13 @@ def create_dispatch(
             ref = ref.name
         if inputs is NotSet:
             inputs = {}
-        status, _, _ = self._requester.requestJson(
-            "POST", f"{self.url}/dispatches", input={"ref": ref, "inputs": inputs}
-        )
-        return status == 204
+        try:
+            self._requester.requestJsonAndCheck("POST", f"{self.url}/dispatches", input={"ref": ref, "inputs": inputs})
+        except GithubException.GithubException:
+            if throw:
+                raise
+            return False
+        return True
```

Comment:
Revisiting this, non-exceptional code paths that use exceptions is bad code design, let's avoid the check if throw is false:
```suggestion
        if throw:
            self._requester.requestJsonAndCheck("POST", f"{self.url}/dispatches", input={"ref": ref, "inputs": inputs})
            return True
        else:
            status, _, _ = self._requester.requestJson(
                "POST", f"{self.url}/dispatches", input={"ref": ref, "inputs": inputs}
            )
            return status == 204
```
This creates an inconsistency for non-204 non-error response codes when `throw == True`, but this is a general issue of this project and can be addressed separately.